server/model/monitor.js

```diff
@@ -383,6 +383,8 @@ class Monitor extends BeanModel {
                     bean.msg = "Monitor under maintenance";
                     bean.status = MAINTENANCE;
                 } else if (this.type === "group") {
+                    let downChildName = "";
+                    let failureCount = 0;
```

Comment:
Continuation of https://github.com/louislam/uptime-kuma/pull/4371#discussion_r1452498689

I can only suggest refactorings for lines you changed => this is why this suggestion is split into multiple parts

My consearn is mostly around future maintenence effort and if this introduces bugs like the faulty (not taking into account `PENDING`) accounting using `failureCount`

```suggestion
                    let downChildNames = [];
```